May-14 14:10:16.047 [main] DEBUG nextflow.cli.Launcher - $> nextflow run python-nextflow.nf -process.echo
May-14 14:10:16.084 [main] INFO  nextflow.cli.CmdRun - N E X T F L O W  ~  version 23.10.1
May-14 14:10:16.098 [main] DEBUG nextflow.plugin.PluginsFacade - Setting up plugin manager > mode=prod; embedded=false; plugins-dir=/Users/srikanth/.nextflow/plugins; core-plugins: nf-amazon@2.1.4,nf-azure@1.3.3,nf-cloudcache@0.3.0,nf-codecommit@0.1.5,nf-console@1.0.6,nf-ga4gh@1.1.0,nf-google@1.8.3,nf-tower@1.6.3,nf-wave@1.0.1
May-14 14:10:16.104 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Enabled plugins: []
May-14 14:10:16.105 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Disabled plugins: []
May-14 14:10:16.107 [main] INFO  org.pf4j.DefaultPluginManager - PF4J version 3.4.1 in 'deployment' mode
May-14 14:10:16.113 [main] INFO  org.pf4j.AbstractPluginManager - No plugins
May-14 14:10:16.165 [main] DEBUG nextflow.cli.CmdRun - Applied DSL=2 by global default
May-14 14:10:16.177 [main] INFO  nextflow.cli.CmdRun - Launching `python-nextflow.nf` [trusting_ekeblad] DSL2 - revision: 925aa6f1a9
May-14 14:10:16.178 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins default=[]
May-14 14:10:16.178 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins resolved requirement=[]
May-14 14:10:16.185 [main] DEBUG n.secret.LocalSecretsProvider - Secrets store: /Users/srikanth/.nextflow/secrets/store.json
May-14 14:10:16.189 [main] DEBUG nextflow.secret.SecretsLoader - Discovered secrets providers: [nextflow.secret.LocalSecretsProvider@692fd26] - activable => nextflow.secret.LocalSecretsProvider@692fd26
May-14 14:10:16.233 [main] DEBUG nextflow.Session - Session UUID: cfb1eea3-9692-4cf1-9228-fa3373f85112
May-14 14:10:16.234 [main] DEBUG nextflow.Session - Run name: trusting_ekeblad
May-14 14:10:16.235 [main] DEBUG nextflow.Session - Executor pool size: 8
May-14 14:10:16.239 [main] DEBUG nextflow.file.FilePorter - File porter settings maxRetries=3; maxTransfers=50; pollTimeout=null
May-14 14:10:16.243 [main] DEBUG nextflow.util.ThreadPoolBuilder - Creating thread pool 'FileTransfer' minSize=10; maxSize=24; workQueue=LinkedBlockingQueue[10000]; allowCoreThreadTimeout=false
May-14 14:10:16.407 [main] DEBUG nextflow.cli.CmdRun - 
  Version: 23.10.1 build 5891
  Created: 12-01-2024 22:01 UTC (13-01-2024 03:31 IDT)
  System: Mac OS X 14.3.1
  Runtime: Groovy 3.0.19 on OpenJDK 64-Bit Server VM 11.0.13+7-b1751.21
  Encoding: UTF-8 (UTF-8)
  Process: 6192@Srikanths-MacBook-Air.local [127.0.0.1]
  CPUs: 8 - Mem: 8 GB (41.1 MB) - Swap: 2 GB (881.9 MB)
May-14 14:10:16.417 [main] DEBUG nextflow.Session - Work-dir: /Users/srikanth/data/scripts/nextflow/work [Mac OS X]
May-14 14:10:16.417 [main] DEBUG nextflow.Session - Script base path does not exist or is not a directory: /Users/srikanth/data/scripts/nextflow/bin
May-14 14:10:16.423 [main] DEBUG nextflow.executor.ExecutorFactory - Extension executors providers=[]
May-14 14:10:16.428 [main] DEBUG nextflow.Session - Observer factory: DefaultObserverFactory
May-14 14:10:16.439 [main] DEBUG nextflow.cache.CacheFactory - Using Nextflow cache factory: nextflow.cache.DefaultCacheFactory
May-14 14:10:16.446 [main] DEBUG nextflow.util.CustomThreadPool - Creating default thread pool > poolSize: 9; maxThreads: 1000
May-14 14:10:16.492 [main] DEBUG nextflow.Session - Session start
May-14 14:10:16.777 [main] DEBUG nextflow.script.ScriptRunner - > Launching execution
May-14 14:10:16.810 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: null
May-14 14:10:16.810 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'local'
May-14 14:10:16.814 [main] DEBUG nextflow.executor.Executor - [warm up] executor > local
May-14 14:10:16.817 [main] DEBUG n.processor.LocalPollingMonitor - Creating local task monitor for executor 'local' > cpus=8; memory=8 GB; capacity=8; pollInterval=100ms; dumpInterval=5m
May-14 14:10:16.819 [main] DEBUG n.processor.TaskPollingMonitor - >>> barrier register (monitor: local)
May-14 14:10:16.893 [main] DEBUG nextflow.Session - Workflow process names [dsl2]: helloworld
May-14 14:10:16.894 [main] DEBUG nextflow.Session - Igniting dataflow network (1)
May-14 14:10:16.894 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > helloworld
May-14 14:10:16.894 [main] DEBUG nextflow.script.ScriptRunner - Parsed script files:
  Script_2b16582fa5ca2c99: /Users/srikanth/data/scripts/nextflow/python-nextflow.nf
May-14 14:10:16.894 [main] DEBUG nextflow.script.ScriptRunner - > Awaiting termination 
May-14 14:10:16.894 [main] DEBUG nextflow.Session - Session await
May-14 14:10:16.976 [Task submitter] DEBUG n.executor.local.LocalTaskHandler - Launch cmd line: /bin/bash -ue .command.run
May-14 14:10:16.977 [Task submitter] INFO  nextflow.Session - [2a/023a15] Submitted process > helloworld
May-14 14:10:17.013 [Task monitor] DEBUG n.processor.TaskPollingMonitor - Task completed > TaskHandler[id: 1; name: helloworld; status: COMPLETED; exit: 127; error: -; workDir: /Users/srikanth/data/scripts/nextflow/work/2a/023a156a497b17a26d094530ae2423]
May-14 14:10:17.015 [Task monitor] DEBUG nextflow.processor.TaskProcessor - Handling unexpected condition for
  task: name=helloworld; work-dir=/Users/srikanth/data/scripts/nextflow/work/2a/023a156a497b17a26d094530ae2423
  error [nextflow.exception.ProcessFailedException]: Process `helloworld` terminated with an error exit status (127)
May-14 14:10:17.024 [Task monitor] ERROR nextflow.processor.TaskProcessor - Error executing process > 'helloworld'

Caused by:
  Process `helloworld` terminated with an error exit status (127)

Command executed:

  #!/usr/bin/python
  print("Hello world")

Command exit status:
  127

Command output:
  (empty)

Command error:
  .command.run: line 92: /usr/bin/python: No such file or directory

Work dir:
  /Users/srikanth/data/scripts/nextflow/work/2a/023a156a497b17a26d094530ae2423

Tip: you can try to figure out what's wrong by changing to the process work dir and showing the script file named `.command.sh`
May-14 14:10:17.027 [main] DEBUG nextflow.Session - Session await > all processes finished
May-14 14:10:17.028 [Task monitor] DEBUG nextflow.Session - Session aborted -- Cause: Process `helloworld` terminated with an error exit status (127)
May-14 14:10:17.037 [main] DEBUG nextflow.Session - Session await > all barriers passed
May-14 14:10:17.038 [Task monitor] DEBUG n.processor.TaskPollingMonitor - <<< barrier arrives (monitor: local) - terminating tasks monitor poll loop
May-14 14:10:17.040 [main] DEBUG n.trace.WorkflowStatsObserver - Workflow completed > WorkflowStats[succeededCount=0; failedCount=1; ignoredCount=0; cachedCount=0; pendingCount=0; submittedCount=0; runningCount=0; retriesCount=0; abortedCount=0; succeedDuration=0ms; failedDuration=16ms; cachedDuration=0ms;loadCpus=0; loadMemory=0; peakRunning=1; peakCpus=1; peakMemory=0; ]
May-14 14:10:17.238 [main] DEBUG nextflow.cache.CacheDB - Closing CacheDB done
May-14 14:10:17.246 [main] DEBUG nextflow.script.ScriptRunner - > Execution complete -- Goodbye
